File: Leetcode/36.validSudoku/36.cpp
class Solution {
public:
bool isValidSudoku(vector<vector<char> >& board)
{
	bool h[9][9] = {false};
	bool v[9][9] = {false};
	bool s[9][9] = {false};
	int i = 0, j = 0;
	for (i = 0; i < 9; ++i)
		for (j = 0; j < 9; ++j)
		{
			if (board[i][j] != '.')
			{
				int t = board[i][j] - '1';
				if (h[i][t])  return false;
				h[i][t] = true;
				if (v[j][t])  return false;
				v[j][t] = true;
				if (s[i / 3 * 3 + j / 3][t]) return false;
				s[i / 3 * 3 + j / 3][t] = true;
			}
		}
return true;
}
};

File: Leetcode/36.validSudoku/36.py
class Solution(object):
   def isValidSudoku(self,board):
		#rows,columns,squares={},{},{}  three dictionary{ 0: set() ,1 :set()....8: set()};
		rows,columns,squares=({i:set() for i in range(9)}
							for _ in range(3))
		for r in range(9):
			for c in range(9):
				num=board[r][c]
				if num=='.':
					continue
				s=(r/3)*3+c/3
				if num in rows[r] or num in columns[c] or num in squares[s]:
					return False
				rows[r].add(num)
				columns[c].add(num)
				squares[s].add(num)
		return True
